apps/dash-oil-gas-ternary/app.py

```diff
@@ -0,0 +1,790 @@
+import importlib
+import pathlib
+import os
+
+import pandas as pd
+import numpy as np
+
+import dash
+import dash_core_components as dcc
+import dash_html_components as html
+import plotly.graph_objs as go
+from dash.dependencies import Input, Output, State
+
+constants = importlib.import_module("apps.dash-oil-gas-ternary.constants")
+
+# app initialize
+app = dash.Dash(__name__)
+server = app.server
+app.config["suppress_callback_exceptions"] = True
+
+# mapbox
+mapbox_access_token = "pk.eyJ1IjoieWNhb2tyaXMiLCJhIjoiY2p1MDR5c3JmMzJsbjQ1cGlhNHA3MHFkaCJ9.xb3lXp5JskCYFewsv5uU1w"
+
+# Load data
+APP_PATH = str(pathlib.Path(__file__).parent.resolve())
+
+df = pd.read_csv(os.path.join(APP_PATH, os.path.join("data", "test_composition.csv")))
+df_prod = pd.read_csv(
+    os.path.join(APP_PATH, os.path.join("data", "YearlyProduction_table_1.csv"))
+)
+
+# Assign color to legend
+colormap = {}
+for ind, formation_name in enumerate(df["fm_name"].unique().tolist()):
+    colormap[formation_name] = constants.colors[ind]
+
+
+def build_banner():
+    return html.Div(
+        id="banner",
+        className="banner",
+        children=[
+            html.Img(src=app.get_asset_url("dash-logo.png")),
```

Comment:
I personally think the dash logo is a little big too big for the desktop version compare to the mockup, but that's up to your discretion